components/WorkSectionAsGallery.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+
+import PropTypes from 'prop-types';
+
+import { ContentfulMedia, SimpleFragment } from 'models';
+
+import { Markdown } from 'components/base';
+import Work from 'components/Work';
+
+import get from 'utils/get';
+
+const renderWorkGalleryAssetBlock = (workGalleryBlock) => {
+  const displayFirstAssetAsFullWidth = get(workGalleryBlock, 'fields.displayFirstAssetAsFullWidth', true);
+  const works = get(workGalleryBlock, 'fields.works', []);
+
+  return (
+    <div className="flex-col">
+      {displayFirstAssetAsFullWidth && 
+        <>
+          <div className="block col-8 relative">
+            <Work work={works[0]}/>
+          </div>
+          <div className="flex col-8 pb_5 md:pb1 items-end">
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[1]}/>
+              </div>
+            </div>
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[2]}/>
+              </div>
+            </div>
+          </div>
+        </>
+      }
+      {!displayFirstAssetAsFullWidth && 
+        <>
+          <div className="flex col-8 pb_5 md:pb1 items-end">
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[0]}/>
+              </div>
+            </div>
+            <div className="block col-4 pr_25 md:pr_5">
+              <div className="relative">
+                <Work work={works[1]}/>
+              </div>
+            </div>
+          </div>
+          <div className="block col-8 relative">
+            <Work work={works[2]}/>
+          </div>
+        </>
+      }
+    </div>
+  );
+}
+
+const renderWorkGalleryTextBlock = (textBlock) => {
+  const text = get(textBlock, 'fields.text', '');
+  return (
+    <div>
+      {text &&
+        <Markdown
+          className="WorkSectionAsGallery__work-gallery-text my4 lg:my10 px1_25 md:px5 lg:px10 Markdown--medium"
+          src={text}
+        />
+      }
+    </div>
+  );
+};
+
+const WorkSectionAsGallery = (props) => {
+  const fields = get(props, 'workGallery.fields', {}); 
+  const workGalleryAssetBlocks = get(fields, 'workGalleryAssetBlocks', []);
+  const workGalleryTextBlocks = get(fields, 'workGalleryTextBlocks', []);
+  const renderAssetBlocksFirst = get(fields, 'renderAssetBlocksFirst', true);
+  const maximumLengthBlocks = workGalleryAssetBlocks.length > workGalleryTextBlocks.length ? [...Array(workGalleryAssetBlocks.length).keys()] : [...Array(workGalleryTextBlocks.length).keys()]; 
```

Comment:
what is this maximumLengthBlocks?

Reply:
I wanted to replicate a for loop here to loop through all the elements of both workGalleryAssetBlocks and workGalleryTextBlocks. the issue, however, is because both may be different sizes, we don't know what the stop condition would be. that's where the logic of finding the one with the longest length comes into play!

once we know which one of them is the longest, we can use that as our stop condition when looping (or, in this case, mapping, since we're using .map()). 

lmk what you think!

Reply:
changed the variable naming + added a comment to make things more clear :)